runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -839,11 +839,18 @@ ClassFileOracle::computeSendSlotCount(U_16 methodIndex)
 			while ((index < count) && ('[' == bytes[index])) {
 				++index;
 			}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			if ((index >= count) || (('L' != bytes[index]) && ('Q' != bytes[index]))) {
+#else
 			if ((index >= count) || ('L' != bytes[index])) {
+#endif /* J9VM_OPT_VALHALLA_VALUE_TYPES */
```

Comment:
J9 prefers a single open brace outside the ifdef to avoid confusing formatting/parsing tools. Given that, the Q case could be added undef ifdef without the need to duplicate the common code:

```
if (common)
#ifdef VT
|| VT code
#endif
{
```

with appropriate formatting.

Reply:
Due to the nature of the conditions, I don't think the condition can be split into `(common)` and `VT code`. This is because the condition needs `(not L) AND (not Q)`.

I can refactor the code to the below code. Is that sufficient?
```
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
			if ((index >= count) || (('L' != bytes[index]) && ('Q' != bytes[index]))) {
#else
			if ((index >= count) || ('L' != bytes[index]))
#endif /* J9VM_OPT_VALHALLA_VALUE_TYPES */
			{
```

Reply:
Another alternative (lets call this 2, and the previous 1):
```
			if (index >= count)
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
				|| (('L' != bytes[index]) && ('Q' != bytes[index]))
#else
				|| ('L' != bytes[index])
#endif /* J9VM_OPT_VALHALLA_VALUE_TYPES */
			{
```
This would keep the `if` outside the `if defined`'s

Reply:
Sorry, didn't read the bracketting properly.

Reply:
I like option 2 better. @tajila ?

Reply:
Im fine with option 2

Reply:
Option 2 has been implemented

Reply:
Looks like code in the format
```
f (index == 1)
	|| (index == 2)
{
```
does not compile.
Adding `()` around the entire conditional fixes this:

```
if ((index == 1)
	|| (index == 2)
) {
```
Fixed option 2 is now (with shortened whitespace)
```
	if ((index >= count)
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
		|| (('L' != bytes[index]) && ('Q' != bytes[index]))
#else
		|| ('L' != bytes[index])
#endif /* J9VM_OPT_VALHALLA_VALUE_TYPES */
	) {
		break;
	}
```